src/components/CopyLinkButton.tsx

```diff
@@ -0,0 +1,56 @@
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { TranslationKeys } from "../i18n/i18next";
+import { generateRoomName, wait } from "../lib";
+import { fetchJWT } from "../rooms";
+import { Button } from "./Button";
+
+export const CopyLinkButton: React.FC = () => {
+  const [buttonText, setButtonText] = useState<TranslationKeys>();
+  const { t } = useTranslation();
+
+  const onCopyLink = async () => {
+    const roomName = generateRoomName();
+    const { url } = await fetchJWT(roomName, true, setButtonText);
+
+    if (!url) {
+      setButtonText("Failed to create meeting room");
+    } else {
+      const absoluteUrl = new URL(url, window.location.href);
```

Comment:
In this case, `url = "//" + window.location.host + "/" + roomName` so this is fine.

Reply:
@bcaller I'm happy to update the code to that style (i.e. concatenating the parts of the url by hand rather than relying on the url constructor) if you prefer

Reply:
No need. It's fine to use the `URL` constructor as long as it isn't being used as a security control to enforce a host.

A bad usage would be using `new URL(userGeneratedUrl, "https://brave.com")` to make sure the url is on brave.com because `new URL("//what/ever, "https://brave.com")` is `https://what/ever`.

My comment was just that I checked what the possible values of the `url` parameter were.